Allow players to view their full /history

tWizT3ddreaMr opened the pull request:
This allows a user to view their own history. I didn't want the same thing to be repeated a third time so i made a method .

I couldn't for the life of me come up with a good name for the method, so I left it at the working name for making sure it all worked.

#326 #556

tWizT3ddreaMr committed:
Self history

Didnt want the same thing to be repeated a third time so i made a method and named it an absolutely wrong name but cant thing of a a good name so i am leaving it with a bad until feedback

tWizT3ddreaMr committed:
imports

Somehow i accidentally merged the imports

tWizT3ddreaMr committed:
Update Command.java

I initially wrote this before I updated the branch "Full History", so it made more sense to have an array.
So I rearranged it to better match the layout in PunishmentManager

tWizT3ddreaMr committed:
Initialize list

I missed that error, but saw in the maven check that it failed because of it

tWizT3ddreaMr committed:
I found a bug

I didnt say true when it was history
so it was just throwing all the errors

Hopefuls commented on core/src/main/java/me/leoko/advancedban/utils/Command.java:
```diff
@@ -18,10 +18,7 @@
-import java.util.Calendar;
-import java.util.GregorianCalendar;
-import java.util.List;
+import java.util.*;
```
Please define the imports required that are relevant for the changes rather than using a wildcard.

Hopefuls commented on core/src/main/java/me/leoko/advancedban/utils/Command.java:
```diff
@@ -281,24 +278,22 @@ public enum Command {
-    HISTORY("ab.history",
-            "\\S+( [1-9][0-9]*)?",
+    HISTORY(null,
+            "\\S+( [1-9][0-9]*)?|\\S+|",
```
Can you clarify this change?

Hopefuls reviewed (changes requested):
Please refer to the comments provided in this review.

tWizT3ddreaMr committed:
Requests

I was trying to do what was asked,  however, I couldn't as the line didn't exist, however this redundant import did

tWizT3ddreaMr commented:
The requested change couldn't happen because it no longer existed likely due to the outdated tag

Hopefuls commented:
Could you revise your changes to the state of development-future? If you wish to continue working on this.

Otherwise I might close this pull request

tWizT3ddreaMr commented:
Let me check out that branch, and I will respond again within a week. I don't imagine it will be difficult, but I don't have the same development and test environment as when I originally worked on it.

tWizT3ddreaMr commented:
Yeah my time over the next week or so hasn't cleared up.
If things change I will submit a new pull request.
